primitives/runtime/src/generic/digest.rs

```diff
@@ -33,19 +33,13 @@ use crate::{
 use sp_core::RuntimeDebug;
 
 /// Generic header digest.
-#[derive(PartialEq, Eq, Clone, Encode, Decode, RuntimeDebug, TypeInfo)]
+#[derive(PartialEq, Eq, Clone, Encode, Decode, RuntimeDebug, TypeInfo, Default)]
 #[cfg_attr(feature = "std", derive(Serialize, Deserialize, parity_util_mem::MallocSizeOf))]
 pub struct Digest {
 	/// A list of logs in the digest.
 	pub logs: Vec<DigestItem>,
 }
 
-impl Default for Digest {
-	fn default() -> Self {
-		Self { logs: Vec::new() }
-	}
-}
-
```

Comment:
It is strange that `derive(Serialize, ...)` is gated on `std`, but `Vec::new()` here was not. But obviously it was not you who introduced that.

Reply:
Good quesiton. The module is not conditionally included. I'm not sure why Vec doesn't cause a problem in no std?

Reply:
going to resolve as this is tangential.